src/map/utils/battleutils.cpp

```diff
@@ -1562,6 +1562,29 @@ namespace battleutils
             return false;
         }
 
+        // Songs cannot be interrupted by physical attacks.
+        if ((SKILLTYPE)PSpell->getSkillType() == SKILL_SNG)
+        {
+            // ShowDebug("Is song, interrupt prevented.\n");
+            return false;
+        }
+
+        // Also cannot interrupt when Aquaveil is active, however the counter is reduced per hit taken.
+        if (PDefender->StatusEffectContainer->HasStatusEffect(EFFECT_AQUAVEIL))
```

Comment:
it doesn't consume aquaveil unless the hit would have interrupted (ie. this needs to be after the random change)